src/platform/logger/syslog_logger.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "syslog_logger.h"
+#include <syslog.h>
+
+namespace mpl = multipass::logging;
+
+namespace
+{
+constexpr auto to_syslog_priority(const mpl::Level& level) noexcept
+{
+    switch (level)
+    {
+    case mpl::Level::trace:
+    case mpl::Level::debug:
+        return LOG_DEBUG;
+    case mpl::Level::error:
+        return LOG_ERR;
+    case mpl::Level::info:
+        return LOG_INFO;
+    case mpl::Level::warning:
+        return LOG_WARNING;
+    }
+    return 42;
+}
```

Comment:
Since this is repeated in both loggers, what about putting this function in a new file? It ca be a method in a class `LinuxLogger`, inheriting from `Logger`? This way, `SyslogLogger` and `JournaldLogger` inherit from `LinuxLogger`.